File: java-sdk/lib/src/main/java/models/shared/GroupMini.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GroupMini
 * A base representation of a group.
**/
public class GroupMini {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("group_type")
    public GroupMiniGroupTypeEnum groupType;
    public GroupMini withGroupType(GroupMiniGroupTypeEnum groupType) {
        this.groupType = groupType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public GroupMini withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public GroupMini withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public GroupMiniTypeEnum type;
    public GroupMini withType(GroupMiniTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/groupmini.go
package shared

type GroupMiniGroupTypeEnum string

const (
	GroupMiniGroupTypeEnumManagedGroup  GroupMiniGroupTypeEnum = "managed_group"
	GroupMiniGroupTypeEnumAllUsersGroup GroupMiniGroupTypeEnum = "all_users_group"
)

type GroupMiniTypeEnum string

const (
	GroupMiniTypeEnumGroup GroupMiniTypeEnum = "group"
)

// GroupMini
// A base representation of a group.
type GroupMini struct {
	GroupType *GroupMiniGroupTypeEnum `json:"group_type,omitempty"`
	ID        *string                 `json:"id,omitempty"`
	Name      *string                 `json:"name,omitempty"`
	Type      *GroupMiniTypeEnum      `json:"type,omitempty"`
}
